dbcon/joblist/tuple-bps.cpp

```diff
@@ -2157,256 +2332,93 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
 
             tplLock.unlock();
 
-            for (i = 0; i < size && !cancelled(); i++)
+            vector<vector<_CPInfo>> cpInfos;
+            // Calculate the work sizes.
+            uint32_t currentThreadPoolSize = jobstepThreadPool.getThreadCount();
+            uint32_t numOfThreads = 1;
+            // FIXME: How to choose the right size of thread pool?
```

Comment:
A complicated thing taking a nature of current CPU into account. There are at least 26 phases in instructions execution pipeline so the number can be as big as 26 * number of system threads available. As of now thread pool size it is 4 * system threads. This 4 is based on a very simple and old description of CPU instructions execution pipeline dated eary 00s.

Reply:
We will get back to this after EM and PP runtimes converge.